cmd/gogio/windowsbuild.go

```diff
@@ -264,53 +237,156 @@ func (f *windowsManifest) encode(w io.Writer) error {
 			<dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
 		</asmv3:windowsSettings>
 	</asmv3:application>
-</assembly>`
-	template, err := template.New("manifest").Parse(t)
+</assembly>`)
 	if err != nil {
 		return err
 	}
 
-	return template.Execute(w, f)
+	var manifest bufferCoff
+	if err := t.Execute(&manifest, v); err != nil {
+		return err
+	}
+
+	b.Coff.AddResource(coff.RT_MANIFEST, 1, &manifest)
+
+	return nil
 }
 
-func (f *windowsResources) encode(w io.Writer) error {
-	const t = `{{if .IconPath}}#define IDI_ICON1 1
-IDI_ICON1 ICON "{{escapePath .IconPath}}"{{end}}
-
-#define IDI_MANIFEST 1
-IDI_MANIFEST 24 "{{escapePath .ManifestPath}}"
-
-#define IDI_VERSION 1
-IDI_VERSION VERSIONINFO
-FILEVERSION     {{.Version}}
-PRODUCTVERSION  {{.Version}}
-FILEFLAGSMASK   0X3FL
-FILEFLAGS       0x0L
-FILEOS          0X40004L
-FILETYPE        0X1L
-FILESUBTYPE     0x0L
-BEGIN
-    BLOCK "StringFileInfo"
-    BEGIN
-        BLOCK "04000400"
-        BEGIN
-            VALUE "ProductVersion", "{{.Version}}"
-            VALUE "FileVersion", "{{.Version}}"
-            VALUE "FileDescription", "{{.Name}}"
-            VALUE "ProductName", "{{.Name}}"
-        END
-    END
-    BLOCK "VarFileInfo"
-    BEGIN
-            VALUE "Translation", 0x0400, 0x0400
-    END
-END`
-	template, err := template.New("rc").Funcs(template.FuncMap{"escapePath": func(s string) string {
-		return strings.Replace(s, `\`, `\\`, -1)
-	}}).Parse(t)
-	if err != nil {
+func (b *windowsBuilder) embedInfo(v windowsResources) error {
+	// https://docs.microsoft.com/pt-br/windows/win32/menurc/vs-versioninfo
+	t := newValue(valueBinary, "VS_VERSION_INFO", []io.WriterTo{
+		// https://docs.microsoft.com/pt-br/windows/win32/api/VerRsrc/ns-verrsrc-vs_fixedfileinfo
+		windowsInfoValueFixed{
+			Signature:      0xFEEF04BD,
+			StructVersion:  0x00010000,
+			FileVersion:    v.Version,
+			ProductVersion: v.Version,
+			FileFlagMask:   0x3F,
+			FileFlags:      0,
+			FileOS:         0x40004,
+			FileType:       0x1,
+			FileSubType:    0,
+			FileDate:       [2]uint32{},
+		},
+		// https://docs.microsoft.com/pt-br/windows/win32/menurc/stringfileinfo
+		newValue(valueText, "StringFileInfo", []io.WriterTo{
+			// https://docs.microsoft.com/pt-br/windows/win32/menurc/stringtable
+			newValue(valueText, "04000400", []io.WriterTo{
```

Comment:
Perhaps name the magic value 04000400 with a nearby const.

Reply:
It's not magic, it's Language (idiom) and the Page (index). It's possible to have multiples "pages" for each language. I added one `v.Language` and `page` to make it easier to understand and easier when it supports multiples languages. 

_Currently, `gogio` doesn't know the idiom, so I choose `Process Default Language` (which is the 0x0400)._